tasks/copy_single.yml

```diff
@@ -6,6 +6,15 @@
   # * `dest`
   # * `tmpname`: Temporary name on the remote host
 
+  - name: Create remote directory
+    become: True
+    file:
+      path: "{{ dest | dirname }}"
+      state: directory
+      owner: "{{ local_ca_owner }}"
+      group: "{{ local_ca_group }}"
+      mode: u+rwx
```

Comment:
Hmm, shouldn't this be configurable? At the moment we only have a default mode for files, however, it seems nicer to introduce a new variable for this. Also, since all other modes are numeric IIRC, can you convert this to 0700?

Reply:
I don't think this is important enough to be configurable. Moreover the permissions are very reasonable: The directory should be readable for the owner because it actually belongs to him. The directory should be writable as I am about to write the key material to it. The directory should be executable because I want to access the key material of course.

Setting it to `0700` has different semantics if the directory already exists and is e.g. world-readable. In this case the role modifies the flags of a directory it might not be responsible for.

Reply:
Aah, so Ansible respects the semantics of `+` on ACLs. TIL.
In that case, I stand corrected and the hardcoded value is okay.